### PR TITLE
docs: add example for guardduty_detector

### PR DESCRIPTION
**Pull request by quixoticmonk**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

@wellsiau-aws 
Added exampled, notes and deprecation notes for the datasources based on
https://docs.aws.amazon.com/guardduty/latest/ug/guardduty-feature-object-api-changes-march2023.html#guardduty-feature-enablement-datasource-relation

### CODE REVIEW

File: docs/resources/guardduty_detector.md [hunk 1]
```diff
@@ -10,6 +9,38 @@ description: |-
 
 Resource Type definition for AWS::GuardDuty::Detector
 
+> [!WARNING] 
+>Deleting this resource is equivalent to "disabling" GuardDuty for an AWS region, which removes all existing findings. You can set the enable attribute to false to instead "suspend" monitoring and feedback reporting while keeping existing data. See the [Suspending or Disabling Amazon GuardDuty documentation](https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_suspend-disable.html) for more information.
```
Comment (wellsiau-aws) [inline docs/resources/guardduty_detector.md:13]: nice!

File: docs/resources/guardduty_detector.md [hunk 2]
```diff
@@ -10,6 +9,38 @@ description: |-
...
+
+## Example Usage
+
+The datasources block is deprecated since March 2023. Use the features block instead and [map each datasources block to the corresponding features](https://docs.aws.amazon.com/guardduty/latest/ug/guardduty-feature-object-api-changes-march2023.html#guardduty-feature-enablement-datasource-relation) block.
```
Comment (wellsiau-aws) [inline docs/resources/guardduty_detector.md:17]: nice!

**Review by wellsiau-aws — APPROVED** [review]
LGTM, thanks for the additional usage-detail

**Comment by marcosentino** [comment]
@quixoticmonk Thank you for the contribution!

**Pull request merged**